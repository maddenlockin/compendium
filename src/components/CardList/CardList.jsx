import React from 'react';
import Card from '../Card/Card';
import PropTypes from "prop-types";

export default function CardList({ cardList }) {
    console.log('test', cardList)
    return (
        <ul aria-label='list of pokemon cards' className= 'cardList'>
            {cardList.map((card) => {
                return (
                    <li key={card.pokemon} className='item'>
                        <Card card={card} />
                    </li>
                );
            })}
        </ul>
    );
}

CardList.propTypes = {
    cardList: PropTypes.arrayOf(
        PropTypes.shape({
            id: PropTypes.number.isRequired,
            pokemon: PropTypes.string.isRequired,
        })
    ).isRequired,
};

