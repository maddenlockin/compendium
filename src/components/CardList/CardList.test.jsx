import { render, screen } from "@testing-library/react"
import CardList from "./CardList";

it('should render a list of pokemon cards', () => {
    render(
        <CardList 
            cardList={[
                {
                    id: 214,
                    pokemon: "murkrow",
                    url_image: "http://assets.pokemon.com/assets/cms2/img/pokedex/full/198.png",
                    type_1: 'dark',
                    type_2: 'flying',
                    shape: 'wings',
                    speed: 91,
                },
                {
                    id: 142,
                    pokemon: "gyarados-mega",
                    url_image: "http://assets.pokemon.com/assets/cms2/img/pokedex/full/130_f2.png",
                    type_1: 'water',
                    type_2: 'dark',
                    shape: 'NA',
                    speed: 81,
                }
            ]}
        />
    );

    const cardList = screen.getByRole('list', { name: 'list of pokemon cards' });
    expect(cardList).toMatchSnapshot();
})