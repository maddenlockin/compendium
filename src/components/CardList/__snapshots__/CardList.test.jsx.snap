// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a list of pokemon cards 1`] = `
<ul
  aria-label="list of pokemon cards"
  class="cardList"
>
  <li
    class="item"
  >
    <figure
      aria-label="pokemon card murkrow"
    >
      <img
        alt="$card.card"
        src="http://assets.pokemon.com/assets/cms2/img/pokedex/full/198.png"
      />
      <div
        class="card-details"
      >
        <h3>
          Card: 
          murkrow
        </h3>
        <p>
          Type 1: 
          dark
        </p>
        <p>
          Type 2: 
          flying
        </p>
        <p>
          Shape: 
          wings
        </p>
        <p>
          Speed: 
          91
        </p>
      </div>
    </figure>
  </li>
  <li
    class="item"
  >
    <figure
      aria-label="pokemon card gyarados-mega"
    >
      <img
        alt="$card.card"
        src="http://assets.pokemon.com/assets/cms2/img/pokedex/full/130_f2.png"
      />
      <div
        class="card-details"
      >
        <h3>
          Card: 
          gyarados-mega
        </h3>
        <p>
          Type 1: 
          water
        </p>
        <p>
          Type 2: 
          dark
        </p>
        <p>
          Shape: 
          NA
        </p>
        <p>
          Speed: 
          81
        </p>
      </div>
    </figure>
  </li>
</ul>
`;
